crates/uv/tests/it/python_find.rs

```diff
@@ -532,6 +532,38 @@ fn python_find_venv() {
         ----- stderr -----
         "###);
     }
+
+    // If we query from outside the project, but specify the project via the --project flag, we find the executable in the project's virtual environment
+    #[cfg(not(windows))]
+    {
+        let child_dir = context.temp_dir.child("child-project");
+        child_dir.create_dir_all().unwrap();
+        let pyproject_toml = child_dir.child("pyproject.toml");
+        pyproject_toml
+            .write_str(indoc! {r#"
+        [project]
+        name = "child-project"
+        version = "0.1.0"
+        requires-python = ">=3.11"
+        dependencies = ["anyio==3.7.0"]
+    "#})
+            .unwrap();
+        uv_snapshot!(context.filters(), context.venv().current_dir(child_dir).arg("-q"), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+
+    ----- stderr -----
+    "###);
```

Comment:
You can use `.assert().success();` instead to avoid an empty snapshot.

We should test `python find --project` _before_ creating the virtual environment. In that case, we should ensure we're respecting the `requires-python` range from the child `pyproject.toml`.

Then, we should test with a virtual environment.

Then, we should also test we respect the `--project` root for `.python-version` file discovery too.